display_list/display_list_image_filter.h

```diff
@@ -665,6 +672,70 @@ class DlColorFilterImageFilter final : public DlImageFilter {
   std::shared_ptr<DlColorFilter> color_filter_;
 };
 
+class DlLocalMatrixImageFilter final : public DlImageFilter {
+ public:
+  explicit DlLocalMatrixImageFilter(const SkMatrix& matrix,
+                                    std::shared_ptr<DlImageFilter> filter)
+      : matrix_(matrix), image_filter_(filter) {}
+  explicit DlLocalMatrixImageFilter(const DlLocalMatrixImageFilter* filter)
+      : DlLocalMatrixImageFilter(filter->matrix_, filter->image_filter_) {}
+  DlLocalMatrixImageFilter(const DlLocalMatrixImageFilter& filter)
+      : DlLocalMatrixImageFilter(&filter) {}
```

Comment:
There is no protection against a null filter in the constructors so everywhere `image_filter_` is used here has to protect against null.